### PR TITLE
[lr_schedular] fix: implement proper min_lr_ratio support in cosine scheduler

### PR DESCRIPTION
**Pull request by 0x404**
### Checklist Before Starting

- [x] Search for similar PR(s).

### What does this PR do?

Fix the cosine learning rate scheduler to properly respect `min_lr_ratio` parameter during both warmup and decay phases.

Update warmup phase to start from `min_lr_ratio` instead of 0, ensure decay phase never goes below `min_lr_ratio`, and add explicit `num_cycles` parameter to scheduler config.

Set default values in configuration files and handle `null` values since in some example yaml config, the `min_lr_ratio` is set to `null`.

### High-Level Design

> Demonstrate the high-level design if this PR is complex.

### Specific Changes

> List the specific changes.

### API

> Demonstrate how the API changes if any.

### Usage Example

> Provide usage example(s) for easier usage.

```python
# Add code snippet or script demonstrating how to use this 
```

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluatuion results, etc.

### Additional Info.

- **Issue Number**: fix https://github.com/volcengine/verl/issues/1376
- **Training**: FSDP
- **Inference**: None

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl?tab=readme-ov-file#contribution-guide).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl?tab=readme-ov-file#code-linting-and-formatting).
- [x] Add `[BREAKING]` to the PR title if it breaks any API.
- [x] Update the documentation about your changes in the [docs](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add CI test(s) if neccessary.

### CODE REVIEW

**Review by vermouth1992 — APPROVED** [review]

**Comment by 0x404** [comment]
https://github.com/volcengine/verl/actions/runs/14830735707/job/41776822625?pr=1400 failed, but seems unrelated to this PR.

**Comment by vermouth1992** [comment]
@ccclyu Shall we merge this?

File: verl/trainer/config/ppo_megatron_trainer.yaml [hunk 1]
```diff
@@ -59,7 +59,8 @@ actor_rollout_ref:
       clip_grad: 1.0
       lr_warmup_steps: -1 # Prioritized. Negative values mean delegating to lr_warmup_steps_ratio.
       lr_warmup_steps_ratio: 0.  # the total steps will be injected during runtime
-      min_lr_ratio: null   # only useful for warmup with cosine
+      min_lr_ratio: 0.0   # only used with cosine lr scheduler, default to 0.0
+      num_cycles: 0.5     # only used with cosine lr scheduler, default to 0.5
```
Comment (ccclyu) [inline verl/trainer/config/ppo_megatron_trainer.yaml]: could we temporally remove two config items that are not used in the megatron trainer? we can add back if we have implemented `lr_scheduler` support in the future PR (welcome!)

Comment (0x404) [reply verl/trainer/config/ppo_megatron_trainer.yaml]: no problem, will update this latter tonight.

**Comment by 0x404** [comment]
revised

**Review by ccclyu — APPROVED** [review]

**Review by vermouth1992 — APPROVED** [review]
